### ISSUE TITLE
Cabal new-copy or call postCopy hook from new-install

### ISSUE THREAD

**Issue by arrowd** [haskell/cabal#5933]
I'm using `cabal new-build` to build and install [git-annex package](https://hackage.haskell.org/package/git-annex). The package has `Custom` build type and have `postCopy` hooks to generate documentation and create `.desktop` file.

Building with `new-build` works fine, but postCopy hooks aren't called during `cabal new-install`, it simply says `Up to date`. There is also no `new-copy` command to make them run.

Is there any way to run these hooks from `cabal-install`?

**Comment by phadej** [comment]
My POV is: if you can live without custom `Setup.sh` you should.
I'm looking at http://hackage.haskell.org/package/git-annex-7.20190219/src/Setup.hs

---

Writing outside package locations (e.g. to users' directory) is very bad practice, please **don't**. Make a subcommand or utility to perform post installation to create `.desktop`.

---

Links creation for `git-annex-shell` works by accident, if it works. Note: the https://github.com/haskell/cabal/pull/5870, `--install-method=copy` wouldn't work as expected, AFAICS.
---

For man pages, wouldn't it be easier for everyone if man pages are pre-processed and included in the`sdist`?

---

See however https://github.com/agda/agda/pull/3563 for similar problem, it's somewhat acceptable as there one **requires** the built `agda` executable to type-check built-ins.

There
- buildHook builds the documentation
- copy and install hooks expand the package description to include newly generated data files

Yet, even there, I'd rather manually type-check those modules in the distribution specific packaging script (say which makes `agda.deb` for Ubuntu), than rely on `Setup.sh` black magic. The problem one tries to solve that, if `agda` is installed as `root` globally in `root`-only writeable place, then builtins have to type-checked by the `root`. For local installs, the whole exercise is not required.

**Comment by arrowd** [comment]
I don't quite get what `Setup.sh` you are talking about. If it is a plain old shell script, how would it work on Windows? Would another `Setup.bat` be needed, then?

I'm expecting from a build system to provide me way to do such simple (from my POV, of course) things.

Anyways, I'm not `git-annex` developer, so I just relay your comments upstream.

**Comment by 23Skidoo** [comment]
> I don't quite get what Setup.sh you are talking about.

It was a typo, @phadej meant `Setup.hs`.

**Comment by arrowd** [comment]
I think this can be closed now? We have a new hooks machinery now and git-annex does not use `postCopy` hook anymore.

**Comment by ulysses4ever** [comment]
@arrowd I agree. Since you're the original author, I don't think we need any ceremony to close it. Anyone doubting this decision should feel free to reopen.
